apps/omg_status/test/omg_status/alert/alarm_printer_test.exs

```diff
@@ -25,68 +27,48 @@ defmodule OMG.Status.Alert.AlarmPrinterTest do
   end
 
   test "if the process has a previous backoff set", %{alarm_printer: alarm_printer} do
-    :erlang.trace(alarm_printer, true, [:receive])
-    %{previous_backoff: previous_backoff} = :sys.get_state(alarm_printer)
-    assert is_number(previous_backoff)
+    assert capture_log(fn ->
+             :erlang.trace(alarm_printer, true, [:receive])
+             %{previous_backoff: previous_backoff} = :sys.get_state(alarm_printer)
+             assert is_number(previous_backoff)
+           end)
   end
 
   test "that the process sends itself a message after startup", %{alarm_printer: alarm_printer} do
-    %{previous_backoff: previous_backoff} = :sys.get_state(alarm_printer)
-    parent = self()
-    :dbg.start()
-    {:ok, _} = :dbg.tracer(:process, {fn msg, _ -> send(parent, msg) end, []})
-    :dbg.p(alarm_printer, :send)
-    :ok = Process.sleep(previous_backoff)
-    result = Enum.count(find_warn_print(alarm_printer, __MODULE__.Alarm.all()))
-    assert result == Enum.count(__MODULE__.Alarm.all())
-  end
+    assert capture_log(fn ->
+             %{previous_backoff: previous_backoff} = :sys.get_state(alarm_printer)
+             :erlang.trace(alarm_printer, true, [:send])
+             :ok = Process.sleep(previous_backoff)
 
-  test "that the process increases the backoff", %{alarm_printer: alarm_printer} do
-    %{previous_backoff: previous_backoff} = :sys.get_state(alarm_printer)
-    assert is_number(previous_backoff)
-    :dbg.stop_clear()
-    parent = self()
-    {:ok, _} = :dbg.tracer(:process, {fn msg, _ -> send(parent, msg) end, []})
-    {:ok, _} = :dbg.p(alarm_printer, [:c])
-    {:ok, _} = :dbg.tp(OMG.Status.AlarmPrinter, :handle_info, 2, [])
-
-    receive do
-      _ ->
-        Process.sleep(previous_backoff)
-    end
-
-    min = previous_backoff
-    max = 3 * previous_backoff
-    %{previous_backoff: new_backoff} = :sys.get_state(alarm_printer)
-    assert min < new_backoff and new_backoff < max
-    :dbg.stop_clear()
-    ### check if backoff is increasing
-    Enum.each(1..5, fn _ ->
-      %{previous_backoff: backoff} = :sys.get_state(alarm_printer)
-      :print_alarms = send(alarm_printer, :print_alarms)
-      min = backoff
-      max = backoff * 3
-      %{previous_backoff: new_backoff} = :sys.get_state(alarm_printer)
-      assert min < new_backoff and new_backoff < max
-    end)
-  end
+             assert_receive {:trace, _, :send, {:notify, {:warn, _, {Logger, "An alarm was raised 1", {_, _}, _}}},
```

Comment:
Epic refactor 🎖

Reply:
yeah, sometimes you need to step away and think again what you did 3 month ago